src/Rules/Functions/ArrayValuesRule.php

```diff
@@ -0,0 +1,88 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rules\Functions;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\FuncCall;
+use PHPStan\Analyser\Scope;
+use PHPStan\Reflection\ReflectionProvider;
+use PHPStan\Rules\Rule;
+use PHPStan\Rules\RuleErrorBuilder;
+use PHPStan\Type\Accessory\AccessoryArrayListType;
+use PHPStan\Type\VerbosityLevel;
+use function count;
+use function sprintf;
+use function strtolower;
+
+/**
+ * @implements Rule<Node\Expr\FuncCall>
+ */
+class ArrayValuesRule implements Rule
+{
+
+	public function __construct(
+		private readonly ReflectionProvider $reflectionProvider,
+		private readonly bool $treatPhpDocTypesAsCertain,
+	)
+	{
+	}
+
+	public function getNodeType(): string
+	{
+		return FuncCall::class;
+	}
+
+	public function processNode(Node $node, Scope $scope): array
+	{
+		if (!($node->name instanceof Node\Name)) {
+			return [];
+		}
+
+		if (AccessoryArrayListType::isListTypeEnabled() === false) {
+			return [];
+		}
+
+		$functionName = $this->reflectionProvider->resolveFunctionName($node->name, $scope);
+
+		if ($functionName === null || strtolower($functionName) !== 'array_values') {
+			return [];
+		}
+
+		$args = $node->getArgs();
+
+		if (count($args) !== 1) {
+			return [];
+		}
+
+		if ($this->treatPhpDocTypesAsCertain) {
+			$arrayType = $scope->getType($args[0]->value);
```

Comment:
To account for named argument call like `array_values(unused: 'foo', array: $array)` you need to normalize this first. You should call `ArgumentsNormalizer::reorderFuncArguments()`. The `$parametersAcceptor` argument you can get by fetching `FunctionReflection` from ReflectionProvider and then doing `ParametersAcceptorSelector::selectFromArgs()` with `$function->getVariants()`. And write a test for this too :)

Reply:
I hope it is correct now. But when there is unsupported named argument (that unused/unsupported argument) - this rule doesn't return any error, because I hope another rule check whether function call is called with correct argument count / names.

Reply:
Yeah sure there is 😊